src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
> Hmm, actually I guess the :JRubyMetricsService essentially was already required via the JRubyPuppetService for the setup work we did around the http-client metrics 

Ah. If the JRubyMetricsService is already a hard dependency of JRubyPuppetService then that makes it much simpler as there isn't a way for folks to leave it out of bootstrap.cfg and still have a bootable Puppet Server.

> One other thing to consider is that not every use of a JRuby instance involves going through the request-handler-service. 

I thought about this a bit as well and I think the handler separation is a good thing at the moment --- as stampeding puppet agents are the problem we want to target and rejecting other traffic such as console or API requests would be a problem. It might be worth enforcing this by adding a check for the Puppet `User-Agent` header.